### PR TITLE
Mnaboka/add config and validator

### PR DESCRIPTION
**Pull request by darkonie**

### CODE REVIEW

**Comment by jdef** [comment]
very nice

File: api/config.go [hunk 1]
```diff
@@ -144,8 +231,72 @@ func LoadDefaultConfig(args []string) (config Config, err error) {
...
+		return defaultConfig, err
+	}
+
+	// validate the result of overriding the default config with values from a config file
+	return defaultConfig, validateConfigStruct(defaultConfig)
+}
+
+func validate(documentLoader gojsonschema.JSONLoader) error {
+	schemaLoader := gojsonschema.NewStringLoader(configJSONSchema)
+	schema, err := gojsonschema.NewSchema(schemaLoader)
```
Comment (malnick) [inline api/config.go]: :+1:

**Comment by malnick** [comment]
This PR is a game changer for how we do config file and CLI flag validation. Totally awesome.

**Comment by malnick** [comment]
Oh yeah, :ship: it

**Pull request merged**